_posts/2023-01-03-xtdb-phoenix.md

```diff
@@ -0,0 +1,304 @@
+then assert that that attribute must be present. As documents themselves are not typed in any way, we
+need to have at least one required attribute that is not present in documents generated from other struct
+types (or we can add a static type attribute with `M.static`).
+
+That is all the definition we need to do to be able to store and query persons. Calling `Person.mapping()`
+will register the mapping for use. Calls to install mappings should be placed in the application startup.
+
+We can now store new Persons by calling `:xt.put/1` like:
+```elixir
+:xt.put(%Person{id: "demo1",
+                first_name: "Dear",
+                last_name: "Reader",
+                email: "dear.reader@example.com"})
+{:ok,{42,{timestamp,1672312695820}}}
+```
+
+That call sends the transaction to the server and waits for a response. Here the server
+responds with ok and a tuple containing the new transaction id and a timestamp.
+
+How about querying the data? That is also simple as xtdberl allows querying by providing a "candidate"
+instance and automatically generates a query that will pull all instances that match the candidate.
+The candidate can have field values (tested with equality) or operations like `{:<, 42}` or `{:textsearch, "D*"}`.
+The comparison operators work on all types, not just numbers, so you can compare things like strings
+and dates as well. The included Lucene `textsearch` operator only works for text.
+
+```elixir
+## Query by specific value
+:xt.ql(%Person{first_name: "Dear"})
+[%Person{id: "demo1",
+         first_name: "Dear",
+         last_name: "Reader",
+         email: "dear.reader@example.com"}]
+
+## Or by an operator, here a Lucene text search
+:xt.ql(%Person{email: {:textsearch, "example"}})
+[...same result as above...]
+```
+
+
+## Putting it all together
+
+With the introduction in place, it's time to put everything together and build our app.
+This section assumes that you have Elixir and Phoenix Framework installed and ready to go.
+You will also need Java (17+) to run the XTDB database.
+
+![Happy family ready for business](/img/2023-xtdb-phoenix/all-together.png)
+
+
+### Create a new app
+
+First, we initialize a new Phoenix app and remember to use the `--no-ecto` parameter. Ecto
+is the Elixir library typically used for interacting with datastores, especially SQL databases.
+Here we don't need it as we use XTDB directly.
+
+```shell
+mix phx.new xthello --no-ecto
+cd xthello
+```
+
+Then we modify `mix.exs` file and include the dependency by adding the
+following line inside deps:
+
+```elixir
+  {:xt, git: "https://github.com/tatut/xtdberl", branch: "main"}
+```
+
+Then we run `mix deps.get` to fetch all the dependencies and we are ready to launch!
+
+Launch an interactive shell and the application by using the command:
+```shell
+iex --erl "-sname xthello@localhost" -S mix phx.server
+```
+
+You should see startup messages and a URL that points you to `http://localhost:4000`.
```

Comment:
What kind of files the initalization did create or what kind of directory structure is present?

I found little bit hard to follow what were autogenerated for the application and what was added.